### PR TITLE
Player movement

### PR DESCRIPTION
**Pull request by Eryngii01**
Implemented 4 player states (WalkingState, RunningState, HidingState, DyingState) to handle the player movement behaviour on Update with both OnEnter and OnExit methods for future extension in terms of adding animations, audio, VFX, etc. 

Also implemented a HRGauge class that is regulated by the PlayerController class to handle all operations regarding increasing and decreasing the heart rate gauge as well as the threshold, which includes the handling of the player death state change.

### CODE REVIEW

File: Assets/Code/Scripts/Player/WalkingState.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        
+        public void OnEnter(PlayerState prevState)
+        {
+            // TODO: When the Player walks...what should happen? music? visual animations? Does it matter from which
+            // state?
+        }
+        
+        public void Movement(Transform player, HRGauge heartRate, Action playerDeath, float baseSpeed)
+        {
+            float speed = baseSpeed + _speedMultiplier;
```
Comment (adashivers) [inline Assets/Code/Scripts/Player/WalkingState.cs]: did you mean to add or multiply here?

Comment (Eryngii01) [reply Assets/Code/Scripts/Player/WalkingState.cs]: Ah I meant to add here but made the variable name a little misleading so I'll adjust that

File: .idea/.idea.to-jam-23/.idea/.gitignore
Comment (eon704) [inline .idea/.idea.to-jam-23/.idea/.gitignore:1]: It looks like this .gitignore should be in the root directory 😅

Comment (eon704) [reply .idea/.idea.to-jam-23/.idea/.gitignore:1]: Also I believe this folder should ignored as well

**Review by eon704 — APPROVED** [review]
.idea folder will need to be removed, otherwise LGTM!

**Pull request merged**